### PR TITLE
micro-opt pass on turf.dm

### PR DESCRIPTION
**Pull request by Iamgoofball**
:cl: Iamgoofball
experimental: Attempts to speed up explosions and singularities.
/:cl:

### CODE REVIEW

File: code/game/turfs/turf.dm [hunk 1]
```diff
@@ -255,12 +264,13 @@
...
 /turf/proc/phase_damage_creatures(damage,mob/U = null)//>Ninja Code. Hurts and knocks out creatures on this turf //NINJACODE
-	for(var/mob/living/M in src)
+	var/list/contents_cached = contents // FUCKING NINJA CODE
+	for(var/mob/living/M in contents_cached)
 		if(M==U)
 			continue//Will not harm U. Since null != M, can be excluded to kill everyone.
 		M.adjustBruteLoss(damage)
 		M.Unconscious(damage * 4)
-	for(var/obj/mecha/M in src)
+	for(var/obj/mecha/M in contents_cached) // REMOVE NINJA CODE
```
Comment (kevinz000) [inline code/game/turfs/turf.dm]: image
kebab

Comment (Iamgoofball) [reply code/game/turfs/turf.dm]: Literally nothing wrong with more remove ninja comments

File: code/game/turfs/turf.dm [hunk 2]
```diff
@@ -108,12 +108,14 @@
 /turf/attackby(obj/item/C, mob/user, params)
 	if(..())
 		return TRUE
+	var/list/contents_cached = contents
 	if(can_lay_cable() && istype(C, /obj/item/stack/cable_coil))
```
Comment (MrStonedOne) [inline code/game/turfs/turf.dm]: You're creating a cached version of this var even if this `if` never passes, seems wasteful.

Also you could likely swap the order here. iirc istype() overhead is less than proc call overhead, you'd want the

Comment (Iamgoofball) [reply code/game/turfs/turf.dm]: Will do

File: code/game/turfs/turf.dm [hunk 3]
```diff
@@ -343,17 +355,19 @@
...
-	for(var/I in src)
+	for(var/I in contents_cached)
 		var/atom/A = I
 		if(ignore_mobs && ismob(A))
 			continue
 		if(ismob(A) || .)
 			A.narsie_act()
 
-/turf/ratvar_act(force, ignore_mobs, probability = 40)
+/turf/ratvar_act(force, ignore_mobs, probability = 40) // todo: merge with above somehow, copypasta
```
Comment (ExcessiveUseOfCobblestone) [inline code/game/turfs/turf.dm]: eldrich_act with another arg being god?

Comment (Iamgoofball) [reply code/game/turfs/turf.dm]: I'd need to refactor stuff, out of scope

Comment (ExcessiveUseOfCobblestone) [reply code/game/turfs/turf.dm]: I was just giving a suggestion based on the comment.

**Comment by optimumtact** [comment]
if your'e going to run optizmiation passes

you should also run profiles you little shit

**Comment by vuonojenmustaturska** [comment]
is "caching" contents even an optimization? var access is not as slow as it used to, and contents is a special type of list with performance benefits over a generic list

**Comment by Cyberboss** [comment]
yeah i'm not convinced, builtins are supposedly already faster to access. Not to mention, using src skips the lookup entirely.

**Comment by KorPhaeron** [comment]
It's a noncompiling goof PR episode

**Comment by Iamgoofball** [comment]
KOR LOOK AT THE COMPILE LOG

MASTER WAS FUCKED

**Comment by Iamgoofball** [comment]
sorry about the all caps, phone keyboard likes to crash if I swap between caps lock/shift/lowercase too much

**Comment by vuonojenmustaturska** [comment]
In short, I don't believe "caching" contents has any measurable performance benefit. I'm willing to be proven wrong, but you'll have to include a profile. I, or another maintainer, will reopen this PR should you do so.

**Pull request closed without merging**